primitives/trie/src/recorder.rs

```diff
@@ -135,6 +158,69 @@ impl<H: Hasher> Recorder<H> {
 		mem::take(&mut *self.inner.lock());
 		self.encoded_size_estimation.store(0, Ordering::Relaxed);
 	}
+
+	/// Stat a new transaction.
+	pub fn start_transaction(&self) {
+		let mut inner = self.inner.lock();
+		inner.accessed_nodes_transactions.push(Default::default());
+		inner.recorded_keys_transactions.push(Default::default());
+	}
+
+	/// Rollback the latest transaction.
+	///
+	/// If there isn't any transaction, nothing gonna happen.
```

Comment:
Maybe a log error message would be reasonable? Otherwise we can silently skip API misuses.

Reply:
I was thinking about getting some information if higher level are not calling `start` and `rollback/commit` in pairs (e.g. due to some error in logic). W/o error message this case will be silently skipped (maybe making future debugging more tricky).

If this is a conscious design decision that rollback/commit can be called w/o matching start than it is also OK.

Logging error could be some compromise between silently ignoring this case and returning the error.

But maybe my reasoning is too defensive here.

Reply:
Done